openwisp_radius/api/serializers.py

```diff
@@ -266,6 +270,41 @@ class Meta:
         read_only_fields = ('organization',)
 
 
+class UserGroupCheckSerializer(serializers.ModelSerializer):
+    result = serializers.SerializerMethodField()
+
+    class Meta:
+        model = RadiusGroupCheck
+        fields = ('attribute', 'op', 'value', 'result')
+
+    def get_result(self, obj):
```

Comment:
I think we'll reuse this logic method in different places, please double check if we can avoid duplication, I think it could be stored in the counter subdir.

Reply:
We only move the following code to the counter base class 

```diff
diff --git a/openwisp_radius/counters/base.py b/openwisp_radius/counters/base.py
index 7c08ab5..2f396c6 100644
--- a/openwisp_radius/counters/base.py
+++ b/openwisp_radius/counters/base.py
@@ -126,6 +126,14 @@ class BaseCounter(ABC):
 
         return int(remaining)
 
+    def result(self):
+        try:
+            return self.check()
+        except MaxQuotaReached:
+            return '0'
+        except SkipCheck:
+            return None
+
 
 class BaseDailyCounter(BaseCounter):
     check_name = 'Max-Daily-Session'

```

We will still need to instantiate the Counter class in the serializer. That's how it is done in FreeRADIUS authorize view. 

https://github.com/openwisp/openwisp-radius/blob/90786b1a9d84d7d9121ca962de0c159e46985551/openwisp_radius/api/freeradius_views.py#L294-L335